### ISSUE TITLE
Problem with 1D + operator

### ISSUE THREAD

**Issue by MUECKE446** [aleph7/Surge#71]
let A = [1.0,2.0,3.0]
        let B = [4.0,5.0,6.0]
        let C = A + B   <-------- error: Ambiguous use of operator '+'
        print(C)
        var D : ValueArray<Double> = [1.0,2.0,3.0]
        var E : ValueArray<Double> = [4.0,5.0,6.0]
        let F = D + E   <-------- this works ok
        print(F)

I have no idea how can I fix it: for a build in Double array + for appending exists.
Can someone help?

**Comment by ghost** [comment]
Yes, for Array `+` is defined as concatenation. You have to use `ValueArray`. You can also write as:
```
var D = ValueArray([1.0,2.0,3.0])
var E = ValueArray([4.0,5.0,6.0])
let F = D + E
```

**Comment by MUECKE446** [comment]
Hello Alejandro,

this is evident, but I mean, that than the public func for + in the package should be removed.

 

Von: Alejandro Isaza [mailto:notifications@github.com] 
Gesendet: Dienstag, 21. März 2017 03:04
An: aleph7/Upsurge
Cc: MUECKE446; Author
Betreff: Re: [aleph7/Upsurge] Problem with 1D + operator (#71)

 

Yes, for Array + is defined as concatenation. You have to use ValueArray. You can also write as:

var D = ValueArray([1.0,2.0,3.0])
var E = ValueArray([4.0,5.0,6.0])
let F = D + E

**Comment by alejandro-isaza** [comment]
There are two options if you want to concatenate:
1. Use `A.append(contentsOf: B)`
2. Do this in a Swift file that does not import Upsurge
